ccl/changefeedccl: TestAlterChangefeedAddTargetsDuringSchemaChangeError failed

ccl/changefeedccl.TestAlterChangefeedAddTargetsDuringSchemaChangeError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11814859?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11814859?buildTab=artifacts#/) on release-23.1 @ [12a0fdf76785787a3a7e83198f1adfd7184ea910](https://github.com/cockroachdb/cockroach/commits/12a0fdf76785787a3a7e83198f1adfd7184ea910):


```
=== RUN   TestAlterChangefeedAddTargetsDuringSchemaChangeError
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestAlterChangefeedAddTargetsDuringSchemaChangeError3667549506
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestAlterChangefeedAddTargetsDuringSchemaChangeError
    alter_changefeed_test.go:1262: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestAlterChangefeedAddTargetsDuringSchemaChangeError3667549506
--- FAIL: TestAlterChangefeedAddTargetsDuringSchemaChangeError (52.62s)
=== RUN   TestAlterChangefeedAddTargetsDuringSchemaChangeError/kafka
    helpers_test.go:823: making server as secondary tenant
    helpers_test.go:901: making kafka feed factory
    alter_changefeed_test.go:1243: condition failed to evaluate within 45s: waiting for checkpoint
    --- FAIL: TestAlterChangefeedAddTargetsDuringSchemaChangeError/kafka (52.61s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedAddTargetsDuringSchemaChangeError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31622

--- a/pkg/ccl/changefeedccl/alter_changefeed_test.go
+++ b/pkg/ccl/changefeedccl/alter_changefeed_test.go
@@ -1170,9 +1170,11 @@ func TestAlterChangefeedAddTargetsDuringSchemaChangeError(t *testing.T) {
 		require.NoError(t, jobFeed.Pause())
 
 		var maxCheckpointSize int64 = 100 << 20
-		// Checkpoint progress frequently, and set the checkpoint size limit.
+		// Ensure that checkpoints happen every time by setting a large checkpoint size.
+		// Because setting 0 for the FrontierCheckpointFrequency disables checkpointing,
+		// setting 1 nanosecond is the smallest possible value.
 		changefeedbase.FrontierCheckpointFrequency.Override(
-			context.Background(), &s.Server.ClusterSettings().SV, 10*time.Millisecond)
+			context.Background(), &s.Server.ClusterSettings().SV, 1*time.Nanosecond)
 		changefeedbase.FrontierCheckpointMaxBytes.Override(
 			context.Background(), &s.Server.ClusterSettings().SV, maxCheckpointSize)
 

